src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -227,6 +229,18 @@ private void GetPackageAndFileDefinitions()
                     _fileDefinitions.Add(fileItem);
                 }
             }
+
+            string GetPackageDiagnosticLevel(LockFileLibrary package)
+            {
+                var messages = LockFile.LogMessages.Where(log => log.LibraryId == package.Name);
```

Comment:
This should probably filter by `TargetGraphs` as well so that diagnostics appear in the correct place on multi-targeting project trees.

I'll investigate adding a new input for the `TargetFrameworkMoniker`.

Reply:
Done in ff326ab49345bb6ee9dc0fc302afa5c02113c385.